### ISSUE TITLE
Configuring API key for Youtube App on SmartTV

### ISSUE THREAD

**Issue by Thomahawkuru** [gabe565/CastSponsorSkip#112]
Hi, First of all thanks for the great project. Been using it for a year now and works great using chromecast 2

Recently our old tv broke and we upgraded to a smart tv. It has built in Youtube which i can cast to from my phone or pc. I was wandering if CastSponsorSKip also works for these kind of devices? 

I read in #70 that it should be supported, but an api key needs to be set for it to work. I'm not sure how this is done correctly. I've generated an api key for an empty project in google cloud, and configured it in the castsponsorskip.service file. CastSponsorSkip runs as a service and detects the chromecast 2, but not the Smart TV.

Any guidance is greatly appreciated. Thanks.

**Comment by Nightwalker83** [comment]
According the section here:
https://github.com/gabe565/CastSponsorSkip/tree/main

The api key is used by the Chrome Cast not youtube so you actually need a ChromeCast to use this script.

**Comment by Thomahawkuru** [comment]
@Nightwalker83 Thanks for taking your time to reply. However, it seems you are referring to the Sponsorblock API? My question is in reference to [CSS_YOUTUBE_API_KEY](https://github.com/gabe565/CastSponsorSkip/tree/main?tab=readme-ov-file#notable-envs), referenced in a comment by @gabe565 in another issue where he states: 

"Do you mean YouTube's TV streaming offering, or the YouTube app on a TV? Because it works with the latter, but you'll need to generate a YouTube API key." https://github.com/gabe565/CastSponsorSkip/issues/70#issuecomment-1970549724

Adding the Youtube API key to CastSPonsorSkip ENVs did not enable it to control the YouTube app on the smartTV for me, while Gabe is implying it should be possible. (My original chromcast device works without setting the youtube api, as expected). 

If someone can confirm whether this is officially supported or unsupported, that would be great!

**Comment by Nightwalker83** [comment]
You don't need the api in order to get it to work in most cases. However, I do not have a smart tv  to test with only a dumb only which works well with the ChromeCast and CSS without the api key needing to be entered. I just put the api key in the file anyway just to be complete. Not sure what is different about the Chromecast 2. All I did with my Chromecast HD was plugged it into power and HDMI on the tv?

**Comment by Thomahawkuru** [comment]
Yes, i am aware. As stated before, my Chromecast V2 works out of the box without setup of youtube API, same as yours. However, CastSponsorSkip does not auto detect my SmartTV's built-in youtube app in the same way, while it is a castable app (from smartphone/chrome browser).

As i read it from Gabe's comment, the youtube API key is needed for a youtube app on a TV. But setting the API does not enable it to work for me on Hisense Vidaa OS. Maybe that only works for devices with android/google tv, like the nvidia shield he is using.

For now i've found a working alternative: [iSponsorblocktv](https://github.com/dmunozv04/iSponsorBlockTV), though it's python.

**Comment by Nightwalker83** [comment]
Ah, I switched to iSponsorblocktv too! I was having issues with CSS not skipping or muting ads despite being detected my Chromecast.

**Comment by gabe565** [comment]
@Nightwalker83 @Thomahawkuru You're right that an API key is required, since Google TV devices don't directly report the video ID, but otherwise it seems to work on my end. 

What type of device are you both using? It works on my Nvidia Shield and Sony Bravia TV, but I don't have any other Google TV devices to test with.

The only other thing that I can think of is maybe your devices are being filtered out? Can you run CSS with `CSS_LOG_LEVEL=debug`? CSS filters out devices that don't report being able to display video...I wonder if your devices don't report that value as I expected.

**Comment by Nightwalker83** [comment]
I am using a Chromecast HD
